ddtrace/profiling/profiler.py

```diff
@@ -202,9 +203,15 @@ def __attrs_post_init__(self):
         exporters = self._build_default_exporters()
 
         if exporters:
-            self._scheduler = scheduler.Scheduler(
-                recorder=r, exporters=exporters, before_flush=self._collectors_snapshot
-            )
+            lambda_function_name = os.environ.get("AWS_LAMBDA_FUNCTION_NAME")
```

Comment:
I'd move that in the "init" part, as a an `attr.ib`
It'd also make it easy to override, if needs be.